contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -82,6 +82,51 @@ library SafeERC20 {
         }
     }
 
+    /**
+     * @dev Perform an {ERC1363} transferAndCall, with a fallback to the simple {ERC20} transfer if the target has no
+     * code. This can be used to implement an {ERC721}-like safe transfer that rely on {ERC1363} checks when
+     * targeting contracts.
+     */
+    function transferAndCallRelaxed(IERC1363 token, address to, uint256 value, bytes memory data) internal {
+        if (to.code.length == 0) {
+            safeTransfer(token, to, value);
+        } else {
+            token.transferAndCall(to, value, data);
+        }
+    }
+
+    /**
+     * @dev Perform an {ERC1363} transferFromAndCall, with a fallback to the simple {ERC20} transferFrom if the target
+     * has no code. This can be used to implement an {ERC721}-like safe transfer that rely on {ERC1363} checks when
+     * targeting contracts.
+     */
+    function transferFromAndCallRelaxed(
+        IERC1363 token,
+        address from,
+        address to,
+        uint256 value,
+        bytes memory data
+    ) internal {
+        if (to.code.length == 0) {
+            safeTransferFrom(token, from, to, value);
+        } else {
+            token.transferFromAndCall(from, to, value, data);
+        }
+    }
+
+    /**
+     * @dev Perform an {ERC1363} approveAndCall, with a fallback to the simple {ERC20} approve if the target has no
+     * code. This can be used to implement an {ERC721}-like safe transfer that rely on {ERC1363} checks when
+     * targeting contracts.
+     */
+    function approveAndCallRelaxed(IERC1363 token, address to, uint256 value, bytes memory data) internal {
+        if (to.code.length == 0) {
+            forceApprove(token, to, value);
+        } else {
+            token.approveAndCall(to, value, data);
+        }
+    }
```

Comment:
In theory no token supporting `approveAndCall` will require approval to be set to 0. But practically it is possible for a token behaving like that to upgrade and add the ERC1363 module. It implies that if there exists an UDST-like token supporting ERC1363, it will revert when calling a contract, but not with an EOA.

Thinking twice about this, I wonder if it makes sense to add mention this inconsistency in the docs:

```suggestion
     * @dev Perform an {ERC1363} approveAndCall, with a fallback to the simple {ERC20} approve if the target has no
     * code. This can be used to implement an {ERC721}-like safe transfer that rely on {ERC1363} checks when
     * targeting contracts.
     *
     * NOTE: If the `to` address has code (i.e. is a contract), this function doesn't deal with tokens requiring approval
     * to be set to zero first (like USDT) before updating the approval.
     */
    function approveAndCallRelaxed(IERC1363 token, address to, uint256 value, bytes memory data) internal {
        if (to.code.length == 0) {
            forceApprove(token, to, value);
        } else {
            token.approveAndCall(to, value, data);
        }
    }
```